### ISSUE TITLE
NET5 ImageList GDI object leak

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/winforms#4928]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/NET5-ImageList-GDI-object-leak/1424122)._

---
In NET5 any Windows Form that contains a System.Windows.Forms.ImageList that is populated from a `((System.Windows.Forms.ImageListStreamer)(resources. GetObject("imageList1.ImageStream")));` Leak GDI objects that are never released by the Garbage Collector, NOT an issue in NET Core 3.1, I have tested in NET6 Preview 3 and it is sill an issue.

Steps to reproduce

1. Create a WinForms application this will default with a form called Form1.
2. Add a second form to the Application called Form2
3. In the Forms Designer add an ImageList to Form2 and Add any Bitmap to the ImageList
4. Add Two buttons to Form1, one to Open Form2 and Open to Garbage Collect
    ```cs
        private void button1_Click(object sender, EventArgs e)
		{
			GC. Collect();
		}

        private void button2_Click(object sender, EventArgs e)
		{
			Form2 oForm = new Form2();
            oForm.Show();
		}
    ```
5. Open Task manger select the details tab and add GDI objects column if not already visible.
6. Run the application and Click Button 2 to open Form2 then close Form2 click Button 1 to garbage collect and you will see GDI object have increased. repeat this and you will see the GDI object increaing with every open and close of Form2

Attached is a sample Solution with the Project reproduced in it.

---
#### Feedback Bot on 5/17/2021, 10:23 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### S Myczko on 5/17/2021, 05:30 PM: 

<p>Looking at this in more detail it appears the the Dispose is never called on the this.imageList1.ImageStream, if I keep a reference to that object and call Dispose()  my self it leak goes away, but this does required a change to the Forms Designer code.</p>
<pre><code>protected override void Dispose(bool disposing)
	{
		if (disposing &amp;&amp; (components != null))
		{
			components.Dispose();
			this.oImageStream.Dispose();
		}
		base.Dispose(disposing);
	}
</code></pre>

---
### Original Solutions
(no solutions)

**Comment by Lydia-Shi** [comment]
**More info:**
It's a regression issue, it occurs in 5.0&6.0, cannot occurs in 3.1&3.0

**Repro Steps:**
1.	Extract the attachment [CoreImageListGDILeak.zip](https://github.com/dotnet/winforms/files/6499537/CoreImageListGDILeak.zip)
2.	Open Task manger, select the 'Details' tab and add GDI objects column if not already visible
3.	Build and run the application, click ‘ShowForm2’ button to open Form2 then close Form2, then click ‘Collect’ button, repeat this operation several times

**Actual behavior:**
The number of GDI objects are increasing.
![Core](https://user-images.githubusercontent.com/54019172/118616608-6dfafa80-b7f4-11eb-9ee2-90815359372d.gif)

**Expected behavior:**
"GDI objects" don't increase.
![Expected](https://user-images.githubusercontent.com/54019172/118617377-26c13980-b7f5-11eb-8970-4ce54926ef18.gif)

**Comment by RussKie** [comment]
I think this may be related to https://github.com/dotnet/winforms/issues/3567

**Comment by kirsan31** [comment]
> I think this may be related to #3567

Yes. it's exactly this issue. It's leak due to unmanned `NativeImageList` destroy was removed from managed `NativeImageList` finalizer...

**Comment by RussKie** [comment]
Duplicate of #3567

**Comment by Zheng-Li01** [comment]
Verified the issue with latest 6.0.0-preview.5.21275.7, the issue has been fixed that "GDI objects" don't increase as expected.
![VerifiedIssue4928](https://user-images.githubusercontent.com/38325459/119796997-6e8a4400-bf0c-11eb-9aea-e8174e9cdc26.gif)

**Comment by StefanMyczko** [comment]
Will this be fixed in NET5 or will we have to wait until NET6 release in November?

**Comment by kirsan31** [comment]
Will be in 5.0.8.

**Comment by Zheng-Li01** [comment]
The issue also have been fixed in 5.0.8.
